### DISCUSSION TITLE
ioSender - a brief tour blog post

### DISCUSSION THREAD

**Discussion by phil-barrett** [terjeio/ioSender#208] [General]
I've written a blog post tour of ioSender.  This is because I continue to get questions and some push back when I recommend it. Hopefully this will help people understand why it is worth trying.  I ran into some difficulty when trying to write the probing section.  Not sure how deep I should dive - currently I think a brief overview is called for.  I will definitely point to Stefan's excellent videos

Any suggestions, ideas, errors are encouraged and welcome.

[https://www.grbl.org/single-post/one-sender-to-rule-them-all](https://www.grbl.org/single-post/one-sender-to-rule-them-all)

**Comment by terjeio** [comment]
Great. Maybe point to [usage tips](https://github.com/terjeio/ioSender/wiki/Usage-tips) as well? Or at least mention how to set the current position via clicking in the DRO field, changing the value and pressing return.

As for push-back I guess that is to be expected, especially from users of other senders as change is something most of us does not like.

**Reply by phil-barrett** [reply]
Thanks.  Done.  it reminded me to link to the Wiki as well.

**Reply by phil-barrett** [reply]
In earlier versions of ioSender, error,  alarm and setting codes were available as CSVs.  Now with localization, they are no longer easily readable.  Perhaps they should be in the Wiki? I would be happy to do it you are ok with it.  At least for English.  Not sure setting codes are needed, though.

**Reply by terjeio** [reply]
> In earlier versions of ioSender, error, alarm and setting codes were available as CSVs.

They still are, used for legacy Grbl controllers. grblHAL has a [number of commands](https://github.com/grblHAL/core/wiki/Report-extensions#code-sets-and-enumerations) available for outputting them, this since plugins (and new versions) may add to the sets making them dynamic. The [config app](https://github.com/terjeio/ioSender/releases/tag/2.0.34) has _File_ menu entries for fetching and saving to a file:

![image](https://user-images.githubusercontent.com/20260062/155831173-4031e265-b81d-490d-ad94-f4b3e35740ea.png)

Is this needed for the senders as well? Or better to present alarm and error messages in on-screen lists?

**Reply by phil-barrett** [reply]
Thanks.  

> Is this needed for the senders as well? Or better to present alarm and error messages in on-screen lists?

Good question.  I think being able to see them from the app makes a lot of sense.
